rre-dataset-generator/dataset_generator/logger.py

```diff
@@ -0,0 +1,114 @@
+"""
+This module uses Python's standard ``logging`` module to emit logs in a JSON format, ensuring the output is machine-readable.
+
+The main components are:
+- ``JsonFormatter``: A custom formatter class that converts log records into
+  a single line of JSON.
+- ``configure_logging``: An idempotent function to set up the root logger with
+  the custom JSON formatter.
+
+"""
+
+from __future__ import annotations
+
+import json
+import logging
+import sys
+from datetime import datetime, timezone
+from typing import Any, Dict
+
+# A set of all standard attributes of a LogRecord.
+# Used by the custom formatter to extract any "extra" data from the record.
+LOGRECORD_STANDARD_ATTRS = {
+    "args", "asctime", "created", "exc_info", "exc_text", "filename",
+    "funcName", "levelname", "levelno", "lineno", "module", "msecs",
+    "message", "msg", "name", "pathname", "process", "processName",
+    "relativeCreated", "stack_info", "thread", "threadName",
+}
+
+
+class JsonFormatter(logging.Formatter):
+    """
+    Formats log records as a single line of JSON.
+    
+    This formatter ensures that logs are machine-readable, which is ideal for
```

Comment:
Why do we have logs formatted to be machine-readable? Our script is not supposed to be monitored by any of the log aggregator tools like ELK, Datadog, or Splunk. We want our logs to be human-readable